contract/contracts/Vesting.sol

```diff
@@ -0,0 +1,49 @@
+pragma solidity ^0.6.0;
+
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "./FanToken.sol";
+
+contract Vesting is Ownable {
+    using SafeMath for uint256;
+
+    mapping(address => bool) public vestingTokens;
+    mapping(address => uint256) public tokensVestingAmount;
+    mapping(address => address) public tokensRecipient;
+    mapping(address => uint256) public tokensVestingStart;
+    mapping(address => uint256) public tokensVestingEnd;
+    mapping(address => uint256) public tokensLastUpdate;
+
+    function addVesting(
+        address token,
+        address recipient,
+        uint256 vestingAmount,
+        uint256 vestingStart,
+        uint256 vestingEnd
+    ) external onlyOwner {
+        require(!vestingTokens[token], "Token is already registered");
+        vestingTokens[token] = true;
+        tokensVestingAmount[token] = vestingAmount;
```

Comment:
I'm wondering if we should set the amount of tokens the address holds so that there is no inconsistency here?

like this `fanToken.balanceOf(address(this));`

Reply:
I dont fully understand but you mean Vesting contract should have a function which returns current token balance?